tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/AadAuthorityTests.cs

```diff
@@ -270,6 +270,7 @@ public void AuthorityCustomPortTest()
 
                 harness.HttpManager.AddSuccessTokenResponseMockHandlerForPost(customPortAuthority);
                 harness.HttpManager.AddInstanceDiscoveryMockHandler(customPortAuthority);
+                harness.HttpManager.AddInstanceDiscoveryMockHandler(customPortAuthority);
 
```

Comment:
this is going to be a problem. we shouldn't need to do instance dis twice.

Reply:
The second instance discovery call should pull from the cache but I wonder how much that will affect perf. I agree it should be avoided if possible

Reply:
i don't remember how this is set-up...why do we need a mock if it's coming from the cache?

Reply:
Maybe it's because if authority has port, we don't cache instance discovery properly? Might need investigation on this?

Reply:
Yes, that is right. This is the only case where instance discovery is not used from cache. Needs more investigation.

Reply:
I still need to investigate this. It is not ok to go to the network twice.

Reply:
Good catch @jennyf19 , appreciate that you review in this level of detail!

So ... the test in its original form is wrong and the change Neha did is wrong :). A network discovery call should happen ONCE and only ONCE. There is a bug however around non-standard clouds that @trwalke is just fixing ([PR](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2760/files))

I will fix the test, but it will continue to "fail" until @trwalke merges his PR.